agent/consul/state/catalog_events_test.go

```diff
@@ -895,7 +1484,8 @@ var cmpPartialOrderEvents = cmp.Options{
 	cmpopts.SortSlices(func(i, j stream.Event) bool {
 		key := func(e stream.Event) string {
 			csn := getPayloadCheckServiceNode(e.Payload)
-			return fmt.Sprintf("%s/%s/%s", e.Topic, csn.Node.Node, csn.Service.Service)
+			// TODO: double check this sort key is correct.
+			return fmt.Sprintf("%s/%s/%s/%s", e.Topic, csn.Node.Node, csn.Service.Service, e.Payload.(EventPayloadCheckServiceNode).key)
```

Comment:
Leaving a comment to call out a TODO, so we double check this is safe and won't hit any problems.

There may be a safer way to do this by using `MatchesKey`